client/MonitoringServiceApiCaller/AutoRegistrationInMonitoring.cs

```diff
@@ -37,10 +36,13 @@ public static class AutoRegistrationInMonitoring
 
             string disableAutoRegistrationStr = configuration[_disableAutoRegistrationEnvVarName];
             if (bool.TryParse(disableAutoRegistrationStr, out bool disableAutoRegistration) && disableAutoRegistration)
+            {
+                log.WriteMonitor("Auto-registration in monitoring", "", $"Auto-registration is disabled");
                 return;
+            }
 
             if (string.IsNullOrWhiteSpace(monitoringServiceUrl))
-                throw new ArgumentException("Argument is empty", nameof(monitoringServiceUrl));
+                throw new ArgumentNullException(nameof(monitoringServiceUrl));
 
             if (log == null)
```

Comment:
Could you move it to the top of the method, please? For it is used above this line, at line 40.